### PR TITLE
[Solidity] Bug fixed for constructor and struct

### PR DESCRIPTION
**Pull request by ChenfengWei0**
```solidity

/////////////// bug 1

Contract x = new Contract(); // ok
Contract x; // segfault

/////////////// bug 2

struct T {
  uint[2] x;
}

T xx = {[1,2]};
xx.x[0]; // wrong value
```

### CODE REVIEW

File: src/solidity-frontend/solidity_convert.cpp [hunk 1]
```diff
@@ -411,6 +411,28 @@ bool solidity_convertert::get_var_decl(
...
+  //  Base x ==> Base x = new Base();
+  else if (
+    SolidityGrammar::get_type_name_t(
+      ast_node["typeName"]["typeDescriptions"]) ==
+    SolidityGrammar::ContractTypeName)
+  {
+    assert(
+      ast_node["typeName"]["nodeType"].get<std::string>() ==
+      "UserDefinedTypeName");
+    const std::string contract_name =
```
Comment (lucasccordeiro) [inline src/solidity-frontend/solidity_convert.cpp:426]: Can you add a comment like "// Extract contract name"?

File: src/solidity-frontend/solidity_convert.cpp [hunk 2]
```diff
@@ -411,6 +411,28 @@ bool solidity_convertert::get_var_decl(
...
+      ast_node["typeName"]["pathNode"]["name"].get<std::string>();
+    exprt val;
+    if (get_implicit_ctor_call(val, contract_name))
```
Comment (lucasccordeiro) [inline src/solidity-frontend/solidity_convert.cpp:432]: Add a comment here too something like "Check if there's an implicit constructor call for the contract"

File: src/solidity-frontend/solidity_convert.cpp [hunk 3]
```diff
@@ -411,6 +411,28 @@ bool solidity_convertert::get_var_decl(
...
+      return true;
+
+    solidity_gen_typecast(ns, val, t);
```
Comment (lucasccordeiro) [inline src/solidity-frontend/solidity_convert.cpp:444]: // Generate typecast for Solidity contract

File: src/solidity-frontend/solidity_convert.cpp [hunk 4]
```diff
@@ -411,6 +411,28 @@ bool solidity_convertert::get_var_decl(
...
+    added_symbol.value = val;
+    decl.operands().push_back(val);
```
Comment (lucasccordeiro) [inline src/solidity-frontend/solidity_convert.cpp:448]: Add constructor call to declaration operands?

File: regression/esbmc-solidity/inheritance_7/test.desc [hunk 5]
```diff
@@ -0,0 +1,4 @@
+CORE
+contract.solast
+--sol contract.sol --k-induction
+^VERIFICATION SUCCESSFUL$
```
Comment (lucasccordeiro) [inline regression/esbmc-solidity/inheritance_7/test.desc:4]: Can you add a version that produces ``VERIFICATION FAILED''?

File: regression/esbmc-solidity/struct_5/test.desc [hunk 6]
```diff
@@ -0,0 +1,4 @@
+CORE
+contract.solast
+--sol contract.sol --k-induction
+^VERIFICATION SUCCESSFUL$
```
Comment (lucasccordeiro) [inline regression/esbmc-solidity/struct_5/test.desc:4]: Can you add a version that produces ``VERIFICATION FAILED''?

File: regression/esbmc-solidity/struct_6/test.desc [hunk 7]
```diff
@@ -0,0 +1,4 @@
+CORE
+contract.solast
+--sol contract.sol --k-induction
+^VERIFICATION SUCCESSFUL$
```
Comment (lucasccordeiro) [inline regression/esbmc-solidity/struct_6/test.desc:4]: Can you add a version that produces ``VERIFICATION FAILED''?

File: regression/esbmc-solidity/struct_7/test.desc [hunk 8]
```diff
@@ -0,0 +1,4 @@
+CORE
+contract.solast
+--sol contract.sol --k-induction
+^VERIFICATION SUCCESSFUL$
```
Comment (lucasccordeiro) [inline regression/esbmc-solidity/struct_7/test.desc:4]: Can you add a version that produces ``VERIFICATION FAILED''?

**Review by lucasccordeiro — CHANGES_REQUESTED** [review]

**Comment by ChenfengWei0** [comment]
Changed as suggested

**Review by lucasccordeiro — APPROVED** [review]
Thanks for submitting this PR, @ChenfengWei0.

**Pull request merged**